### PR TITLE
Unfucks supermatter extraction

### LINKED ISSUE #34730: Supermatter shards can be cut infinitely from the SM at no loss to stability.
Cutting a piece of the SM off will only activate the SM if it wasn't already active. You can, however, cut away at it dozens of times and hoard a ton of shards without any problem to then bap people with for cheeky instant kills.

It should probably significantly drop the stability when you cut a piece off.

### PR DESCRIPTION
**Pull request by DaxDupont**
[Changelogs]: 

:cl: Dax Dupont
fix: People can no longer extract 300 supermatter shard slivers without it exploding.
fix: Fixed thongs not dusting people on use.
/:cl:

[why]: I was bored while cooking supper
Fixes #34730 by doing about 30% damage to the SM. They have about 10 seconds to scram before it warns people. 

Also fixes a runtime, for some reason nobody passed the user, that has now changed and a sanity check was added.
This will now dust people on attack, which might be a balance change. This seems like the intended goal of the code, unless it was just a sloppy copy pasta, please input @KorPhaeron.

### CODE REVIEW

**Comment by DaxDupont** [comment]
If it shouldn't dust them let me know, it just seem like it was intended to, but i can just remove it out right if maintainers want it.

**Comment by DaedalusGame** [comment]
you can probably still get infinite shards if you rcd the floor under the sm but that's a different issue

**Comment by Robustin** [comment]
So someone can blow the SM in 5 seconds just by spamming the scalpel?

Make a spam check on this please.

**Comment by Cyberboss** [comment]
Just give the thing a limited number of uses

**Comment by DaedalusGame** [comment]
Actually, the crowbar snaps in half

**Comment by DaxDupont** [comment]
That works too, mostly went on wjohn suggested. 
@DaedalusGame rcding the floor will make it delam fast.

**Comment by DaedalusGame** [comment]
*rcding the floor will make it unable to delam

**Comment by DaxDupont** [comment]
It checks for vacuums now.

**Comment by DaedalusGame** [comment]
It does not.

**Comment by DaxDupont** [comment]
@DaedalusGame https://github.com/DaxDupont/tgstation/blob/0787155588f0a725d8dc0f887b608192a7d96f6e/code/modules/power/supermatter/supermatter.dm#L275

**Comment by DaedalusGame** [comment]
yes, that's the problem. The explosion code is much further down. This prevents any kind of processing if it's on a space turf.

**Comment by DaxDupont** [comment]
Huh, you're right.

**Comment by Hatfish** [comment]
>Fixed thongs not dusting people on use

I know ERP is illegal but shouldn't you let admins deal with this kind of thing instead of coding in solutions?

File: code/game/objects/items/theft_tools.dm [hunk 1]
```diff
@@ -203,6 +203,7 @@
 	toolspeed = 0.5
 	damtype = "fire"
 	usesound = 'sound/weapons/bladeslice.ogg'
+	var/usesLeft = 2
```
Comment (Jalleo) [inline code/game/objects/items/theft_tools.dm]: Shouldn't this be slightly random to prevent meta?

**Comment by DaxDupont** [comment]
What meta? It's not supposed to be a secret, just a hard limit.

File: code/game/objects/items/theft_tools.dm [hunk 2]
```diff
@@ -249,11 +250,12 @@
 	else
 		investigate_log("has consumed [AM].", "supermatter")
 		qdel(AM)
-	user.visible_message("<span class='danger'>As [user] touches \the [AM] with \a [src], silence fills the room...</span>",\
+	if (user)
+		user.visible_message("<span class='danger'>As [user] touches \the [AM] with \a [src], silence fills the room...</span>",\
 			"<span class='userdanger'>You touch \the [AM] with \the [src], and everything suddenly goes silent.</span>\n<span class='notice'>\The [AM] flashes into dust, and soon as you can register this, you do as well.</span>",\
 			"<span class='italics'>Everything suddenly goes silent.</span>")
+		user.dust()
 	radiation_pulse(user, 500, 2)
```
Comment (ShizCalev) [inline code/game/objects/items/theft_tools.dm]: doesn't dust destroy the mob?
would this still pulse?

File: code/modules/power/supermatter/supermatter.dm [hunk 3]
```diff
@@ -538,12 +538,17 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
...
 		to_chat(user, "<span class='notice'>You carefully begin to scrape \the [src] with \the [W]...</span>")
 		if(do_after(user, 60 * W.toolspeed, TRUE, src))
-			to_chat(user, "<span class='notice'>You extract a sliver from \the [src]. \The [src] begins to react violently!</span>")
-			new /obj/item/nuke_core/supermatter_sliver(drop_location())
-			matter_power += 200
+			if (scalpel.usesLeft)
+				to_chat(user, "<span class='notice'>You extract a sliver from \the [src]. \The [src] begins to react violently!</span>")
+				new /obj/item/nuke_core/supermatter_sliver(drop_location())
+				matter_power += 200
+				scalpel.usesLeft -= 1
```
Comment (ShizCalev) [inline code/modules/power/supermatter/supermatter.dm]: scalpel.usesLeft-- instead

**Review by ShizCalev — DISMISSED** [review]

File: code/game/objects/items/theft_tools.dm (hunk 1 above)
Comment (Cyberboss) [reply code/game/objects/items/theft_tools.dm]: yeah do 2-4

**Comment by ExcessiveUseOfCobblestone** [comment]
is fixes/closes/etc. case sensitive?

WHY ARE THERE NO SQUIGGLY LINES UNDER FIX

**Comment by Cyberboss** [comment]
Make sure you put #<PR number> and not the full url

File: code/modules/power/supermatter/supermatter.dm [hunk 4]
```diff
@@ -538,12 +538,17 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
...
 	if(istype(W, /obj/item/scalpel/supermatter))
+		var/obj/item/scalpel/supermatter/scalpel = W
 		playsound(src, W.usesound, 100, 1)
 		to_chat(user, "<span class='notice'>You carefully begin to scrape \the [src] with \the [W]...</span>")
 		if(do_after(user, 60 * W.toolspeed, TRUE, src))
-			to_chat(user, "<span class='notice'>You extract a sliver from \the [src]. \The [src] begins to react violently!</span>")
-			new /obj/item/nuke_core/supermatter_sliver(drop_location())
-			matter_power += 200
+			if (scalpel.usesLeft)
+				to_chat(user, "<span class='notice'>You extract a sliver from \the [src]. \The [src] begins to react violently!</span>")
```
Comment (Cyberboss) [inline code/modules/power/supermatter/supermatter.dm]: remove `\the`s

File: code/modules/power/supermatter/supermatter.dm [hunk 5]
```diff
@@ -538,12 +538,17 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
...
+				new /obj/item/nuke_core/supermatter_sliver(drop_location())
+				matter_power += 200
+				scalpel.usesLeft--
+			else 
+				to_chat(user, "<span class='notice'>You fail to extract a sliver from \the [src]. \The [W] isn't sharp enough anymore!</span>")
```
Comment (Cyberboss) [inline code/modules/power/supermatter/supermatter.dm]: remove `\the`s

File: code/modules/power/supermatter/supermatter.dm [hunk 6]
```diff
@@ -538,12 +538,17 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
...
 		if(do_after(user, 60 * W.toolspeed, TRUE, src))
-			to_chat(user, "<span class='notice'>You extract a sliver from \the [src]. \The [src] begins to react violently!</span>")
-			new /obj/item/nuke_core/supermatter_sliver(drop_location())
-			matter_power += 200
+			if (scalpel.usesLeft)
+				to_chat(user, "<span class='notice'>You extract a sliver from \the [src]. \The [src] begins to react violently!</span>")
+				new /obj/item/nuke_core/supermatter_sliver(drop_location())
+				matter_power += 200
+				scalpel.usesLeft--
+			else
```
Comment (Cyberboss) [inline code/modules/power/supermatter/supermatter.dm]: notify them somehow on the last use

File: code/game/objects/items/theft_tools.dm [hunk 7]
```diff
@@ -249,11 +254,12 @@
 	else
 		investigate_log("has consumed [AM].", "supermatter")
 		qdel(AM)
-	user.visible_message("<span class='danger'>As [user] touches \the [AM] with \a [src], silence fills the room...</span>",\
+	if (user)
+		user.visible_message("<span class='danger'>As [user] touches \the [AM] with \a [src], silence fills the room...</span>",\
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: remove \the

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**